modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp

```diff
@@ -31,6 +31,9 @@
 #include <sofa/core/behavior/MultiVec.h>
 #include <sofa/simulation/Node.h>
 
+#include <thread>
+#include <functional>
```

Comment:
Hi, thank you very much for this kind of contribution @EulalieCoevoet !
It is funny because yesterday I was thinking of reworking the Gauss Seidel code to parallelize it. 

Back to your PR I think the clean approach would be to enable the multi-thread plugin from SOFA which give access to a MT api where you "just" need to create a runner and give him some work.
I will be happy to help you on that and test your changes. Could you describe which scene you used to test and how did you log it.

Reply:
Hi, thanks @epernod sounds good, I wasn't aware of this plugin, I'll have a look. 
Well in my scene I have 38 constraints per FEM model (700nodes). For the lod I just used `Log Time` of the `Stats` tab of runSofa..